### ISSUE TITLE
Pass structures by reference

### ISSUE THREAD

**Issue by aleixalcacer** [Blosc/c-blosc2#221]
I think it may not be entirely safe to pass some structs (i.e. `blosc_storage`) by value to functions.

As can be seen in `blosc2.h`,  `blosc_storage` is defined as:

```c
typedef struct {
    bool sequential;
    //!< Whether the chunks are sequential (frame) or sparse.
    char* urlpath;
    //!< The path for persistent storage. If NULL, that means in-memory.
    blosc2_cparams* cparams;
    //!< The compression params when creating a schunk.
    //!< If NULL, sensible defaults are used depending on the context.
    blosc2_dparams* dparams;
    //!< The decompression params when creating a schunk.
    //!< If NULL, sensible defaults are used depending on the context.
} blosc2_storage;
```

if this structure is passed by value a copy of the storage will be created. However, the `cparams`, the `dparams` and the `urlpath` of the two storage objects will point to the same memory area.

Is this correct?

**Comment by FrancescAlted** [comment]
Agreed.  As a first step for fixing this, here it is the list of (public) functions that are getting a `storage` as value:

```
blosc2_schunk* blosc2_schunk_new(blosc2_storage storage);
blosc2_schunk* blosc2_schunk_empty(int nchunks, blosc2_storage storage);
blosc2_schunk* blosc2_schunk_copy(blosc2_schunk *schunk, blosc2_storage storage);
blosc2_schunk* blosc2_schunk_open(blosc2_storage storage);
```

We need to convert the `blosc2_storage storage` to `blosc2_storage* storage`, and make everything work again.

But as we are in the middle of a big API refactorization (branch `private-frames`), let's wait after merging the new API.

**Comment by aleixalcacer** [comment]
Fixed in #235
